motionplan/motionPlanner.go

```diff
@@ -509,21 +509,24 @@ func CheckPlan(
 		}
 
 		// get checkFrame's currentInputs
+		// *currently* it is guaranteed that a relative frame will constitute 100% of a solver frame's dof
 		checkFrameCurrentInputs, err := sf.mapToSlice(currentInputs)
 		if err != nil {
 			return err
 		}
+		interpolateConfig := []frame.Input{
+			{Value: checkFrameGoalInputs[0].Value},
+			{Value: checkFrameGoalInputs[1].Value},
+			{Value: checkFrameCurrentInputs[2].Value},
+			{Value: checkFrameGoalInputs[3].Value},
+		}
 
 		// pre-pend to segments so we can connect to the input we have not finished actuating yet
 		segments = append(segments, &ik.Segment{
 			StartPosition: poses[wayPointIdx-1],
 			EndPosition:   poses[wayPointIdx],
-			StartConfiguration: []frame.Input{
-				{Value: checkFrameGoalInputs[0].Value},
-				{Value: checkFrameGoalInputs[1].Value},
-				{Value: checkFrameCurrentInputs[2].Value},
-			},
-			EndConfiguration: checkFrameGoalInputs,
+			StartConfiguration: interpolateConfig,
```

Comment:
[q]: how does interpolation across inputs work if `StartConfiguration == EndConfiguration`?

Reply:
Since the configurations now have start and end distance, the start distance is "where you are now" and the end distance is where you are going. Thus only one set of inputs is needed to interpolate.

We require both to match to prevent errors.